code/modules/tgui/tgui_input_number.dm

```diff
@@ -88,14 +92,17 @@
 	var/timeout
 	/// The title of the TGUI window
 	var/title
+	/// Whether the submitted number is rounded down into an integer.
+	var/round_value
 
 
-/datum/tgui_input_number/New(mob/user, message, title, default, max_value, min_value, timeout)
+/datum/tgui_input_number/New(mob/user, message, title, default, max_value, min_value, timeout, round_value)
 	src.default = default
 	src.max_value = max_value
 	src.message = message
 	src.min_value = min_value
 	src.title = title
+	src.round_value = round_value
```

Comment:
`round_value` should default to `TRUE` if no value is supplied, as the current behavior does

Reply:
Actually I've decided it doesn't really matter